### PR TITLE
Fix/cv2-3693: cannot restore or mark as not spam within items

### PR DESCRIPTION
**Pull request by danielevalverde**
## Description

Re-adding: restore from trash/mark as not spam" button on item page and removing deprecated code related to "confirm" items

Reference: CV2-3693

## Type of change

- [ ] Performance improvement and/or refactoring (non-breaking change that keeps existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Security mitigation or enhancement
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Automated test (add or update automated tests)

## How has this been tested?

- Send an item to the trash
- Go to the item page
Check that the button to "Restore From Trash" is displayed 
- click on the button
Check that the item is restored 

- Mark an item as Spam
- Go to the item page
Check that the button to "Mark as Not Spam" is displayed 
- click on the button
Check that the item is restored 

## Checklist

- [X] I have performed a self-review of my own code
- [ ] I've made sure my branch is runnable and given good testing steps in the PR description
- [ ] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [ ] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I implemented any new components, they are self-contained, their `propTypes` are declared and they use React Hooks and, if data-fetching is required, they use Relay Modern with fragment containers
- [ ] I have removed the /* eslint-disable @calm/react-intl/missing-attribute */ from any files I've worked on and added a `description` prop to all `<FormattedMessage />` components that were missing it.
- [ ] To the best of my knowledge, any new styles are applied according to the design system
- [ ] If I added a new external dependency, I included a rationale for doing so and an estimate of the change in bundle size (e.g., checked in https://bundlephobia.com/)

### CODE REVIEW

File: src/app/components/media/RestoreProjectMedia.js [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+            <FormattedMessage
+              id="mediaActionsBar.movedSpamBack"
+              defaultMessage="Item marked as Not Spam"
+              description="Success message after marking item as Not Spam."
+            />
+          );
+        setFlashMessage(message, 'success');
+        window.location.assign(window.location.pathname);
+      },
+      // FIXME: Add onFailure handler
```
Comment (amoedoamorim) [inline src/app/components/media/RestoreProjectMedia.js:77]: Please add an onFailure handler

File: src/app/components/media/RestoreProjectMedia.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
+/* eslint-disable relay/unused-fields */
```
Comment (amoedoamorim) [inline src/app/components/media/RestoreProjectMedia.js:1]: Please remove this directive and weed out unused query fields

**Review by amoedoamorim — CHANGES_REQUESTED** [review]
Looks great, Dani!! Please address these two comments for a :100:

**Review by brianfleming — COMMENTED** [review]
@danielevalverde not sure if I am missing a detail, but I noticed there is a type of item that I can send to spam, but not restore on the item page:

### In Spam
Screenshot 2023-09-19 at 1 53 53 PM

### Item page, no restore button:
Screenshot 2023-09-19 at 1 53 59 PM

Is there a reason that I am overlooking?

**Comment by brianfleming** [comment]
@danielevalverde Same comment for the trash, same item when moved to the trash cannot be restored from the item page

File: src/app/components/media/RestoreProjectMedia.js [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+import React from 'react';
+import PropTypes from 'prop-types';
+import { createFragmentContainer, graphql } from 'react-relay/compat';
+import Relay from 'react-relay/classic';
+import { FormattedMessage } from 'react-intl';
+import Button from '@material-ui/core/Button';
```
Comment (brianfleming) [inline src/app/components/media/RestoreProjectMedia.js:7]: Could you update this to the cds ButtonMain component please

**Review by brianfleming — CHANGES_REQUESTED** [review]

**Comment by amoedoamorim** [comment]
> @danielevalverde not sure if I am missing a detail, but I noticed there is a type of item that I can send to spam, but not restore on the item page:

I remember seeing code for displaying the restore button specifically for Parent or Main items only.
I think Child / Similar media would be sent to trash / restored via the main item.
That being a suggested similar item, I'm not sure how it should behave.

**Comment by brianfleming** [comment]
> > @danielevalverde not sure if I am missing a detail, but I noticed there is a type of item that I can send to spam, but not restore on the item page:
> 
> I remember seeing code for displaying the restore button specifically for Parent or Main items only. I think Child / Similar media would be sent to trash / restored via the main item. That being a suggested similar item, I'm not sure how it should behave.

Interesting. I think the best thought then would be to try to confirm if this is still true. As a user, if I can send something to the trash/spam, I think I should be able to restore it the same way. So unless we discover a specific reason, I say we remove that restriction

**Comment by danielevalverde** [comment]
@amoedoamorim and  @brianfleming , thanks for all the feedbacks :) 
Done: 
use ButtonMain instead of Button from MaterialUI :white_check_mark: 
remove unused query fields :white_check_mark: 
add onFailure handler :white_check_mark: 
remove the restriction to show "restore from Trash/ mask as Not Spam" when the item is a suggestion :white_check_mark: 

Please review it when you have some time. Thank you :pray:

File: src/app/components/media/RestoreProjectMedia.js [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+          (
+            <FormattedMessage
+              id="mediaActionsBar.failedmovedSpamBack"
+              defaultMessage="Failed to mark the item as Not Spam. Please try again later."
+              description="Failure message after attempting to mark an item as Not Spam."
+            />
+          );
+        setFlashMessage(message, 'failure');
+        window.location.assign(window.location.pathname);
+        setFlashMessage('Failed to restore the item. Please try again later.', 'error');
```
Comment (amoedoamorim) [inline src/app/components/media/RestoreProjectMedia.js:94]: Please review this: use the translatable message with the correct message type (I think 'error')

**Review by brianfleming — APPROVED** [review]
Thanks for the awesome changes @danielevalverde 
spam

File: src/app/components/media/RestoreProjectMedia.js [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+      <FormattedMessage
+        id="mediaActionsBar.notSpamTo"
+        defaultMessage="Not spam"
+        description="Label for button that marks an item as not spam"
+      />
+    );
+  }
+
+  return (
+    <React.Fragment>
```
Comment (amoedoamorim) [inline src/app/components/media/RestoreProjectMedia.js:121]: React.Fragment around a single component seems unnecessary

File: src/app/components/media/RestoreProjectMedia.js [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+    slug: PropTypes.string.isRequired,
+  }).isRequired,
+  projectMedia: PropTypes.shape({
+    id: PropTypes.string.isRequired,
+    dbid: PropTypes.number.isRequired,
+  }).isRequired,
+};
+
+export default createFragmentContainer(RestoreProjectMedia, {
+  team: graphql`
```
Comment (amoedoamorim) [inline src/app/components/media/RestoreProjectMedia.js:148]: is this fragment used anywhere?

**Review by amoedoamorim — CHANGES_REQUESTED** [review]
Dani, I left a couple of notes, please review them! :pray:

**Review by amoedoamorim — APPROVED** [review]
:lgtm: :100:

**Pull request merged**